### DISCUSSION TITLE
How to log atom-in-atoms correctly?

### DISCUSSION THREAD

**Discussion by Atunez** [pmndrs/jotai#2384] [Q&A] [answered]
Hey,

I have been looking around the discussions and docs, but I can't seem to find a good way to log atoms-in-atoms.

Here is the original use case I have:

I have something of this type:
```
export type FlagAtomType = PrimitiveAtom<Flag>;
export type FlagsAtomType = Record<string, FlagAtomType>;
export const flagsAtom = atom<FlagsAtomType>({});
```

I then, later on, add things to the flagsAtom, 

```
const createFlagAtom = (initialFlag: Flag): FlagAtomType => atom(initialFlag);
const newFlagAtom = createFlagAtom(localFlag);
setFlags((flags) => ({
    ...flags,
    [localFlag.name]: newFlagAtom,
}));
```

Then later on, when a button is clicked, I want to log all the current atoms stored... 

(This part is in a different file/component)

```
  function displayAtom() {
   Object.entries(flags).map(([flagName, flagAtom]) =>{
      console.log(flagName, flagAtom);
    });
  }
```

However, the behavior I see is that while the map contains the correct keys, those keys map to the first value the atom had. Which, I think makes sense? From my understanding, we would only get the correct internal value once we use `useAtom` on the inner `flagAtom`.

However, is there a good way to print that atom without having to make a new component to essentially just print it? I am a bit new to react, so I am not sure if there is something I can use that I didn't notice. Possibly something with higher-order hooks?

Thanks for all the help!

**Comment by dai-shi** [comment, marked as answer]
```js
const displayAllFlagsAtom = atom(null, (get) => {
  Object.entries(get(flagsAtom)).forEach(([flagName, flagAtom]) => {
    console.log(flagName, get(flagAtom));
  });
});

  // in a component
  const displayAllFlags = useSetAtom(displayAllFlagsAtom);
  useEffect(() => {
    displayAllFlags();
  }, [displayAllFlags]);
```

**Reply by Atunez** [reply]
Thanks for the quick reply!

Interesting use of the write pattern. Works pretty well :)
